File: leetcode/0509_fibonacci_number.cpp
// Copyright (c) 2019 Jacopo Notarstefano

#include <cassert>

using namespace std;

class Solution {
 public:
  int fib(int N) {
    int current = 0, next = 1;

    for (int i = 0; i < N; i++) {
      int tmp = next;
      next = current + next;
      current = tmp;
    }

    return current;
  }
};

int main() {
  auto solution = Solution();

  assert(1 == solution.fib(2));
  assert(2 == solution.fib(3));
  assert(3 == solution.fib(4));
}

File: leetcode/0509_fibonacci_number.kt
package leetcode

class Solution {
    fun fib(N: Int): Int {
        var current = 0
        var next = 1

        repeat(N) {
            val tmp = next
            next = current + next
            current = tmp
        }

        return current
    }
}

fun main(args: Array<String>) {
    val solution = Solution()

    assert(1 == solution.fib(2))
    assert(2 == solution.fib(3))
    assert(3 == solution.fib(4))
}
